lib/Alien/Base/FAQ.pod

```diff
@@ -0,0 +1,279 @@
+=head1 NAME
+
+Alien::Base::FAQ - Frequently Asked Questions about Alien::Base
+
+=head1 SYNOPSIS
+
+ perldoc Alien::Base::FAQ
+
+=head1 DESCRIPTION
+
+This document serves to answer the most frequently asked questions made by L<Alien::Base> authors.
+
+=head2 What is Alien and Alien::Base?
+
+Alien is a Perl namespace for defining dependencies in CPAN for libraries and tools which are not "native"
+to CPAN.  For a manifesto style description of the Why, and How see L<Alien>.  L<Alien::Base> is a base
+class and framework for creating Alien distributions.  The idea is to address as many of the common challenges
+to developing Alien modules in the base class to simplify the process.
+
+=head2 How do determine cflags and libs for packages that do not support pkg-config?
+
+TODO
+
+=head2 How do I test my package once it is built (before it is installed)?
+
+There are many ways to test Alien modules before they are installed.  Probably the easiest is by using
+L<Test::CChecker>.  Here is an example test for the theoretical libfoo library:
+
+ use Test::More tests => 1;
+ use Test::CChecker 0.06;    # require version 0.06 for compatability with recent Alien::Base
+ use Alien::Base    0.18;    # require version 0.18 for staged install to blib
+ use Alien::Foo;
+ 
+ compile_with_alien 'Alien::Foo';
+ 
+ compile_run_ok <<SOURCE, 'basic compile test for libfoo';
+ #include <foo.h>
+ 
+ int main()
+ {
+   foo();
+   return 0;
+ }
+ SOURCE
+
+=head2 How do I patch packages that need minor (or major) alterations?
+
+One approach is to create a unified diff for patches that you want to apply and simply run patch on them.  For 
+example, your C<Build.PL> might look like this:
+
+ # Build.PL
+ use Alien::Base::ModuleBuild;
+ Alien::Base::ModuleBuild->new(
+   ...
+   alien_build_commands => [
+     # --binary allows patches to be applied on either Unix or Windows
+     'patch --binary -p1 < ../../patch/mypackage.patch',
```

Comment:
iirc, some versions of `patch` (e.g., non-GNU ones on *BSD), dont' support the `--binary` flag